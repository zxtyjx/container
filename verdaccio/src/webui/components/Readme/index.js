
import React from 'react';
import PropTypes from 'prop-types';

import 'github-markdown-css';

const Readme = (props) => {
  return <div className="markdown-body" dangerouslySetInnerHTML={{__html: props.readMe}}/>;
};

Readme.propTypes = {
  readMe: PropTypes.string.isRequired
};

export default Readme;
