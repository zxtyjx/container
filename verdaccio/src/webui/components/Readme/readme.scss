@import '../../styles/variable';

.searchBox {
  width: 100%;
  font-size: 18px;
  line-height: 30px;
  border: none;
  border-bottom: 1px solid lightgrey;
  outline: none;

  &:focus {
    border-bottom: 1px solid grey;
  }
}
