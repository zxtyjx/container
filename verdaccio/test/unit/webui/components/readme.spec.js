/**
 * Readme component
 */

import React from 'react';
import { shallow } from 'enzyme';
import Readme from '../../../../src/webui/components/Readme/index';

console.error = jest.fn();

describe('<Readme /> component', () => {
  it('should give error for the required fields', () => {
    shallow(<Readme />);
    expect(console.error).toBeCalled();
  });

  it('should dangerously set html', () => {
    const props = {
      readMe: '<h1>This is a test string</h1>'
    };
    const wrapper = shallow(<Readme {...props} />);
    expect(wrapper.html()).toEqual(
      '<div class="markdown-body"><h1>This is a test string</h1></div>'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
