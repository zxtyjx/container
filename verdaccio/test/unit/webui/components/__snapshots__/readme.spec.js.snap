// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Readme /> component should dangerously set html 1`] = `"<div class=\\"markdown-body\\"><h1>This is a test string</h1></div>"`;
